internal/controllers/clusterextension_controller.go

```diff
@@ -392,7 +392,7 @@ func (r *ClusterExtensionReconciler) resolve(ctx context.Context, ext ocv1alpha1
 	}
 
 	if ext.Spec.UpgradeConstraintPolicy != ocv1alpha1.UpgradeConstraintPolicyIgnore && installedBundle != nil {
-		upgradePredicate, err := SuccessorsPredicate(installedBundle)
+		upgradePredicate, err := SuccessorsPredicate(ext.Spec.PackageName, installedBundle)
```

Comment:
Could be covered separately/later, but I wonder if the successors predicate should actually care about the package name?

I could envision a separate predicate that gets to decide which package names are acceptable to be upgraded from, where the default would be what it is now (i.e. "spec.packageName").

Something like that may make it easier for us if we wanted to help users do an upgrade between bundles that come from a different package.

Reply:
But perhaps `upgradeConstraintPolicy: Ignore` is enough of an escape hatch for that situation....

Reply:
+1 on covering later. I think there is some additional nuance here of deciding the ordering of bundles between different packages. How would you decide between a newer version of a bundle in the same package versus a different one?

I think for now the current escape hatch should work?